be/src/olap/tablet_schema.h

```diff
@@ -646,6 +663,11 @@ class TabletSchema : public MetadataAdder<TabletSchema> {
     // key: unique_id of column
     // value: extracted path set and sparse path set
     std::unordered_map<int32_t, PathsSetInfo> _path_set_info_map;
+
+    // key: field_pattern
+    // value: index
+    using PatternToIndex = std::unordered_map<std::string, TabletIndexPtr>;
+    std::unordered_map<int32_t, PatternToIndex> _index_by_unique_id;
```

Comment:
what about multi index in same pattern?
 rename `_index_by_unique_id` to `_index_by_unique_id_with_pattern`